### PR TITLE
Create CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by steven-murray**
Initial crack at a contributing guidelines file.

Interested to see whether @jdbowman thinks it is too ambitious/restrictive. Also, it should probably get some fleshing out for C development.

### CODE REVIEW

**Comment by jdbowman** [comment]
Thanks for setting this up!  It is definitely ambitious compared to our previous efforts.  We should think about the core set of rules that will be appropriate but not too onerous.  (I noticed one place where it still says pyuvdata.)

**Comment by steven-murray** [comment]
Thanks @jdbowman . My plan is to make one of the repositories fully follow the guidelines, and then refer to that as an example. Most of the guidelines are actually automatically enforceable and therefore sound more onerous than they actually are. 

Typically, the guidelines are followed by setting up various config files which can usually be copy-pasted.

We can even create an explicit template repository, which I admit I would only know how to do for python repos, but that might cover the majority of cases anyway. 

If you feel that there is just too much in the CONTRIBUTING.md, we could do something like decide on a small core set of rules/guidelines and put them up-front, then provide more detail down in the document? 

BTW, to formally review, you can hit the "Add your review" button. You can then comment on any specific line, and explicitly approve/require modifications/reject the PR.

**Comment by steven-murray** [comment]
@jdbowman to get us all used to the system, I'm waiting for an explicit Review Approval from you before merging!

**Review by jdbowman — APPROVED** [review]

**Review by jdbowman — APPROVED** [review]

**Review by jdbowman — APPROVED** [review]

**Pull request merged**
